physicalplan/function/function.go

```diff
@@ -1,101 +1,279 @@
 // Copyright (c) The Thanos Community Authors.
 // Licensed under the Apache License 2.0.
 
-package scan
+package function
 
 import (
+	"context"
 	"fmt"
 	"math"
+	"sort"
+	"sync"
 	"time"
 
 	"github.com/efficientgo/core/errors"
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/prometheus/prometheus/promql"
 	"github.com/prometheus/prometheus/promql/parser"
 
+	"github.com/thanos-community/promql-engine/physicalplan/model"
 	"github.com/thanos-community/promql-engine/physicalplan/parse"
 )
 
 var InvalidSample = promql.Sample{Point: promql.Point{T: -1, V: 0}}
 
+type FunctionArgs struct {
+	Labels       labels.Labels
+	Points       []promql.Point
+	StepTime     time.Time
+	SelectRange  time.Duration
+	ScalarPoints []promql.Point
```

Comment:
Can we make this:
```suggestion
	ScalarPoints []float64
```

I don't think we can specify points in PromQL functions.

Reply:
So this is basically data points we get from the scalar arguments of a function like `clamp(instant vector, scalar, scalar)`. But yes this can be float64. I'll update!